src/core/lib/security/credentials/composite/composite_credentials.h

```diff
@@ -56,6 +56,12 @@ class grpc_composite_channel_credentials : public grpc_channel_credentials {
     return inner_creds_->update_arguments(args);
   }
 
+  int cmp(const grpc_channel_credentials* other) const override {
+    // TODO(yashykt): Check if we can do something better here
```

Comment:
I think this should delegate to comparing the underlying channel and call creds.  Note that this will require also adding a `cmp()` method to `grpc_call_credentials`.  Once you do that, the code here can be something like this:

```
int r = strcmp(type(), other->type());
if (r != 0) return r;
auto* o = static_cast<grpc_composite_channel_credentials*>(other);
r = inner_->cmp(o->inner_.get());
if (r != 0) return r;
return call_creds_->cmp(o->call_creds_.get());
```

Also, note that basically every implementation will need those first two lines, so we should consider finding a way to make that happen automatically as part of the structure.  For example, maybe wrapping `cmp()` in a parent method that does that check and then calls `cmp()` only if the types are the same.  We would just need to make sure that it's clear which method is implemented by subclasses and which one is called by the external caller, so that we don't get that confused.  (That may be a little ugly; I'm open to other suggestions as well.)

Reply:
I agree that this is what we should be doing here but my original intention was to make this PR a simple change and then look at each TODO item and fix it for each in a focused way so that we don't make any mistakes.

Reply:
I'm fine with filling in the implementations for each creds type in a separate PR.  However, I think we should add the `cmp()` method to `grpc_call_credentials` as part of this PR.